### PR TITLE
Added membership_plans. Fixes #14

### LINKED ISSUE #14: Membership Fees
Hi,

I propose to add a new top-level property called `membership_fees`. It should be an array with objects in it. The properties in the object should be:

- `name`
   - type: string
   - description: For example Student Membership, Normal Membership etc...
- `value`
   - type: number
   - description: How much does it cost!
- `currency`
   - type: string
   - description: In which currency is it
- `renewal_interval`
   - type: string with enum
   - description:  How long does the membership last or when do you have to pay again
   - possible values: yearly, monthly, weekly, biyearly, semiyearly
- `description`
   - type: string
   - description: A free form string

### PR DESCRIPTION
**Pull request by the-metalgamer**

### CODE REVIEW

**Comment by the-metalgamer** [comment]
Any update on this?

**Comment by rnestler** [comment]
@dbrgn, @dns2utf8 any opinions? I currently don't have time to look at it. Is this related to #14 ?

File: 14-draft.json [hunk 1]
```diff
@@ -1132,6 +1132,48 @@
...
+                },
+                "renewal_interval": {
+                    "description": "When does the membership be renewed!",
+                    "type": "string",
+                    "enum": [
+                        "yearly",
+                        "monthly",
+                        "weekly",
+                        "daily",
+                        "hourly"
```
Comment (dbrgn) [inline 14-draft.json]: Maybe this should include "other". What if there is biyearly payment or some other type of system?

If renewal interval is "other", then the description must be filled out. (I don't think that's something we can enforce with JSON schema though. But we can document it.)

Comment (the-metalgamer) [inline 14-draft.json]: Might be a good idea! We add it now!

File: 14-draft.json [hunk 2]
```diff
@@ -1132,6 +1132,49 @@
           "end"
         ]
       }
+    },
+    "membership_plans": {
+        "description": "A list of the different memberhsip plans your hackerspace might have.",
```
Comment (dbrgn) [inline 14-draft.json]: *membership

File: 14-draft.json [hunk 3]
```diff
@@ -1132,6 +1132,49 @@
...
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "description": "The name of the membership plan. For example Student Membership, Normal Membership etc...",
+                    "type": "string"
+                },
+                "value": {
+                    "description": "How much does this plan cost!",
```
Comment (dbrgn) [inline 14-draft.json]: replace `!` with `?`

File: 14-draft.json [hunk 4]
```diff
@@ -1132,6 +1132,49 @@
...
+                    "type": "number"
+                },
+                "currency": {
+                    "description": "What's the currency? It should be formatted according to <a href=\"https://en.wikipedia.org/wiki/ISO_4217\" target=\"_blank\">ISO 4217</a> short-code format.",
+                    "type": "string"
+                },
+                "renewal_interval": {
+                    "description": "When does the membership be renewed! If you select other, please specify in the description what your renewal interval is.",
```
Comment (dbrgn) [inline 14-draft.json]: Rewrite the first sentence to to `How often is the membership renewed?`

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,3 +13,4 @@ Changes should start with one of the following tags:
 - [added] The `contact.keymasters` array items can also contain a `xmpp` field
 - [changed] The value in `sensors.account_balance` can now be any ISO 4217 string
 - [added] The `timezone` key was added under `location`
+- [added] The `membership_plans` was added to represent membership plans a space might have
```
Comment (dbrgn) [inline CHANGELOG.md]: Can you add "key" between "membership_plans" and "was"?

**Comment by the-metalgamer** [comment]
Any update on this?

**Comment by rnestler** [comment]
> Any update on this?

Sorry for the delay! One doesn't get a notification from Github when you push a commit to a branch. So to avoid us not noticing when you add the requested changes, just add a comment which mentions the reviewer who requested them (@dbrgn in this case).

**Comment by dbrgn** [comment]
Looks good to me now! @rnestler, agree?

File: 14-draft.json [hunk 6]
```diff
@@ -1132,6 +1132,49 @@
...
       }
+    },
+    "membership_plans": {
+        "description": "A list of the different membership plans your hackerspace might have.",
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "description": "The name of the membership plan. For example Student Membership, Normal Membership etc...",
```
Comment (rnestler) [inline 14-draft.json]: Minor nitpick: I would use either etc. or ...:
* "The name of the membership plan. For example Student Membership, Normal Membership, ..."
* "The name of the membership plan. For example Student Membership, Normal Membership, etc."

File: 14-draft.json [hunk 7]
```diff
@@ -1132,6 +1132,49 @@
...
+                },
+                "currency": {
+                    "description": "What's the currency? It should be formatted according to <a href=\"https://en.wikipedia.org/wiki/ISO_4217\" target=\"_blank\">ISO 4217</a> short-code format.",
+                    "type": "string"
+                },
+                "renewal_interval": {
+                    "description": "How often is the membership renewed? If you select other, please specify in the description what your renewal interval is.",
+                    "type": "string",
+                    "enum": [
+                        "yearly",
```
Comment (rnestler) [inline 14-draft.json]: Is "yearly" or "annually" more common?

File: 14-draft.json [hunk 8]
```diff
@@ -1132,6 +1132,49 @@
...
+                        "monthly",
+                        "weekly",
+                        "daily",
+                        "hourly",
+                        "other"
```
Comment (rnestler) [inline 14-draft.json]: Do we now which intervals are most common? Maybe add bi-annually? Also at coredump for example we communicate 10CHF/month, but actually bill yearly and also membership is renewed yearly. How would we describe that?
```JSON
{
  "name": "Student",
  "value": 10,
  "currency": "CHF",
  "renewal_interval": "monthly",
  "description": "Billed annually"
}
```
or
```JSON
{
  "name": "Student",
  "value": 120,
  "currency": "CHF",
  "renewal_interval": "yearly"
}
```

I just see some potential for confusion here.

File: 14-draft.json (hunk 7 above)
Comment (dbrgn) [inline 14-draft.json]: I think i would stick with "yearly" since it's more consistent with the other options.

File: 14-draft.json (hunk 8 above)
Comment (dbrgn) [inline 14-draft.json]: I guess the renewal and billing usually match, so we should probably chose the yearly renewal interval in our case.

Comment (rnestler) [inline 14-draft.json]: I think we just need to document clearly, how it is intended to be used.

Comment (dbrgn) [inline 14-draft.json]: Yes, absolutely :) Any proposals for wording? Maybe `How often is the membership renewed (billed)?`?

Comment (rnestler) [inline 14-draft.json]: I think we could just add some concrete examples to the description string?

Comment (the-metalgamer) [inline 14-draft.json]: I would also go with the second example. It's more clear than the first one.

Comment (dbrgn) [inline 14-draft.json]: @the-metalgamer ok, do you want to do the necessary adjustments?

Comment (the-metalgamer) [inline 14-draft.json]: @dbrgn Yes, but currently I'm thinking about how to incorporate the examples in the descriptions

Comment (rnestler) [inline 14-draft.json]: We can also merge it as is and improve it later, to not stall this PR for such a long time.

Comment (the-metalgamer) [inline 14-draft.json]: That might be a good idea!

File: 14-draft.json (hunk 6 above)
Comment (the-metalgamer) [inline 14-draft.json]: Fixed!

File: 14-draft.json (hunk 8 above)
Comment (the-metalgamer) [inline 14-draft.json]: ping @rnestler @dbrgn

**Comment by dbrgn** [comment]
Thanks!

**Pull request merged**